Spec/RealtimeClientConnection.swift

```diff
@@ -2126,6 +2126,33 @@ class RealtimeClientConnection: QuickSpec {
 
                         expect(channel.state).to(equal(ARTRealtimeChannelState.Detached))
                     }
+
+                    it("should resume the connection after an auth renewal") {
+                        let options = AblyTests.commonAppSetup()
+                        options.disconnectedRetryTimeout = 1.0
+                        options.useTokenAuth = true
+                        let client = AblyTests.newRealtime(options)
+                        defer { client.dispose(); client.close() }
+                        let channel = client.channels.get("test")
+
+                        expect(client.connection.state).toEventually(equal(ARTRealtimeConnectionState.Connected), timeout: testTimeout)
```

Comment:
This is not correct, this is not the behaviour we'd expect as a channel ATTACH that does not become ATTACHED would be resent or timeout.

The process I said we needed to test is:

* Connect
* Attach a channel (i.e wait for it to be attached)
* Token expires and connection is disconnected by Ably
* Resume happens automatically, wait until connected
* Publish a message over REST on the channel
* Ensure the message is received on the channel and the channel is still attached